### DISCUSSION TITLE
Syntax Error: Missing Period

### DISCUSSION THREAD

**Discussion by paulegg1** [microsoft/finops-toolkit#621] [Q&A] [answered]
Hi All

So I've download the template and run through a successful deployment.   I have the Storage Account, Key Vault and the ADF.  All seems well and everything looks complete.   No deployment errors.

I've run a cost export and I can see the .csv files under 'msexports'.   At first my trigger didn't fire, but I read the troubleshooting section and noted that it needed to be started. 

The trigger is now started but I have a remaining issue.  The ETL_ingestion pipeline has two syntax errors on "Set Scope" and "Set Date".  The errors seem almost identical - each one complains of a missing period within the expression.  It makes no sense to me and (of course) inserting a period at the suggested position just causes a new syntax error.

Here's the expression from Set Scope:

`@replace(split(pipeline().parameters.folderName,variables('folderArray')[sub(length(variables('folderArray')), if(greater(length(variables('folderArray')[sub(length(variables('folderArray')), 2)]), 12), 3, 4))])[0],'msexports','ingestion')`

The error:

` Position 196 Syntax error: Missing period
`

Position 196 is immediately after the square bracket that follows the '2'.  

Has anyone else seen this and/or have any ideas what might be wrong?  I really want to get this working but at the moment this is a total brick wall for me.

Many thanks,

Paul Eggleton

**Comment by paulegg1** [comment]
Hi All
Apologies, despite searching before I seem to have missed this:

https://github.com/microsoft/finops-toolkit/pull/574

which looks like it includes a fix for 

https://github.com/microsoft/finops-toolkit/issues/557

I will try this out,
Paul.

**Reply by flanakin** [reply]
Sorry you're hitting this @paulegg1. Please let us know if this resolves the issue for you! We're working on finalizing the 0.2.1 release now as we run thru the last remaining open issues. Stay tuned!

**Reply by paulegg1** [reply]
Hi Michael,
Thanks for the reply - yes, this fixed the issue for me.  I will indeed stay tuned for the next release, we are already exporting our costs and beginning to work with the reports in Power BI - this is all really good stuff!

Paul.

**Comment by flanakin** [comment, marked as answer]
FYI - This is now available in 0.3. We decided to bump the version given all the things in the release.
